main.py

```diff
@@ -99,10 +100,14 @@ def main():
 
         # search and add the track into new album
         logging.info(f'Adding tracks to {name}')
-        
+
         for track in playlist.get('tracks'):
+            artist = None
+            title = None
             try:
                 track_data = track.get('track')
+                if not track_data:
+                	track_data = [item for item in library if item['id'] == track['trackId']][0]
```

Comment:
If track metadata doesn't exist from playlist data, go find the track in the library and get its information instead.